src/widgets/settingspages/GeneralPageView.hpp

```diff
@@ -197,14 +197,30 @@ class GeneralPageView : public QWidget
             },
             this->managedConnections_);
 
-        QObject::connect(
-            combo, QOverload<const int>::of(&QComboBox::currentIndexChanged),
-            [combo, &setting,
-             setValue = std::move(setValue)](const int newIndex) {
-                setting = setValue(DropdownArgs{combo->itemText(newIndex),
-                                                combo->currentIndex(), combo});
-                getIApp()->getWindows()->forceLayoutChannelViews();
-            });
+        if (listenToActivated)
+        {
+            QObject::connect(
+                combo, &QComboBox::activated,
+                [combo, &setting,
+                 setValue = std::move(setValue)](const int index) {
+                    setting = setValue(DropdownArgs{
+                        combo->itemText(index), combo->currentIndex(), combo});
+                    getIApp()->getWindows()->forceLayoutChannelViews();
+                });
+        }
+        else
+        {
+            QObject::connect(
+                combo,
+                QOverload<const int>::of(&QComboBox::currentIndexChanged),
+                [combo, &setting,
+                 setValue = std::move(setValue)](const int newIndex) {
+                    setting =
+                        setValue(DropdownArgs{combo->itemText(newIndex),
+                                              combo->currentIndex(), combo});
+                    getIApp()->getWindows()->forceLayoutChannelViews();
+                });
```

Comment:
Aren't these the same callbacks? I think it would be easier to read if the callback was only declared once, like `updateValue` or `applyValue`.

Reply:
Fixed in [`8787e31` (#5224)](https://github.com/Chatterino/chatterino2/pull/5224/commits/8787e3161fc31ee8a7c8df6ad60e66eac499647f)